src/fides/api/alembic/migrations/versions/2736c942faa2_property_specific_messaging_db_models.py

```diff
@@ -0,0 +1,126 @@
+"""property specific messaging db models
+
+Revision ID: 2736c942faa2
+Revises: efddde14da21
+Create Date: 2024-05-28 14:26:09.012859
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "2736c942faa2"
+down_revision = "efddde14da21"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "messaging_template_to_property",
+        sa.Column("id", sa.String(length=255), nullable=False),
+        sa.Column(
+            "created_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column(
+            "updated_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column("messaging_template_id", sa.String(), nullable=False),
+        sa.Column("property_id", sa.String(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["messaging_template_id"],
+            ["messaging_template.id"],
+        ),
+        sa.ForeignKeyConstraint(
+            ["property_id"],
+            ["plus_property.id"],
+        ),
+        sa.PrimaryKeyConstraint("id", "messaging_template_id", "property_id"),
+        sa.UniqueConstraint(
+            "messaging_template_id",
+            "property_id",
+            name="messaging_template_id_property_id",
+        ),
+    )
+    op.create_index(
+        op.f("ix_messaging_template_to_property_id"),
+        "messaging_template_to_property",
+        ["id"],
+        unique=False,
+    )
+    op.create_index(
+        op.f("ix_messaging_template_to_property_messaging_template_id"),
+        "messaging_template_to_property",
+        ["messaging_template_id"],
+        unique=False,
+    )
+    op.create_index(
+        op.f("ix_messaging_template_to_property_property_id"),
+        "messaging_template_to_property",
+        ["property_id"],
+        unique=False,
+    )
+    op.add_column(
+        "messaging_template", sa.Column("is_enabled", sa.Boolean(), nullable=True)
+    )
+    op.execute("UPDATE messaging_template SET is_enabled = FALSE;")
```

Comment:
Wait why do existing messaging templates get set to disabled? Wouldn't they already be in use?

Reply:
Further, your data migration downgrade deletes templates which are disabled, which deletes everything!

Reply:
We're applying the concept of enable/disable by template only to property-specific templates. 

This means we ignore this enable/disable flag for our Basic Templates avail to OSS users, or to Paid users who have not yet enabled the property-specific messaging features.

We set all to disabled so that the user needs to explicitly "enable" (and thus configure) the templates they wish to use for property-specific messaging.

Reply:
still, your downgrade deletes all disabled templates!

Reply:
oh I see!

Reply:
adding a unique prefix to messaging templates I've created and deleting based on this